components/scream/src/share/field/field_boundedness_check.hpp

```diff
@@ -0,0 +1,76 @@
+#ifndef SCREAM_FIELD_BOUNDEDNESS_CHECK_HPP
+#define SCREAM_FIELD_BOUNDEDNESS_CHECK_HPP
+
+#include "share/field/field.hpp"
+#include "ekat/util/ekat_math_utils.hpp"
+
+namespace scream
+{
+
+// This field property check returns true if all data in a given field are
+// bounded by the given upper and lower bounds (inclusively), and false if not.
+// It can repair a field that fails the check by clipping all unbounded values
+// to their closest bound.
+template<typename ScalarType, typename Device>
+class FieldBoundednessCheck: public FieldPropertyCheck<ScalarType, Device> {
```

Comment:
Very minor: BoundednessCheck makes me think we only check that the field is bounded (i.e., finite). Maybe `FieldBoundsCheck` instead?

Reply:
The boundedness check sets upper and lower bounds on the values in the field, which seems to me to be the same as what you think it means. A bounds check makes me think about checking that an index is valid for a field. Am I understanding you correctly?

Reply:
My math mindsetting interpret "boundedness check" as "is the field finite or does it have singularities somewhere?". But that's probably my mindsetting. As I said, this was very minor. And you're the native speaker, so if you think Boundedness makes sense, it works for me too. :)

Reply:
The finite/singularities question is relevant, for sure. From a "boneheaded" physicist's perspective, though, I would think that "a <= x <= b for all x" also implies boundedness of that type, for a and b finite. But I guess you're observing that my version of "boundedness" requires someone to supply the a and b, whereas yours does not.

What about FieldWithinIntervalCheck, then?

You've corrected my English in technical contexts before, so no need to defer on those grounds. :-)